dotCMS/src/main/java/com/dotcms/security/secret/SecretsStore.java

```diff
@@ -0,0 +1,77 @@
+package com.dotcms.security.secret;
+
+import java.util.Optional;
+import com.dotmarketing.util.Config;
+import io.vavr.control.Try;
+import java.util.Set;
+
+public interface SecretsStore {
+
+    /**
+     * verifies if the key is present
+     * @param variableKey
+     * @return
+     */
+     boolean containsKey(final String variableKey);
+
+    /**
+     * Optionally returns the secrets value if there is one, empty if not
+     * 
+     * @param variableKey
+     * @return
+     */
+    Optional<char[]> getValue(final String variableKey);
+
+    /**
+     * Lists the keys - keys can be case insensitive (forced lowercase) based on the implementation
+     * 
+     * @return
+     */
+    Set<String> listKeys();
+
+    /**
+     * deletes all the entries in the secretsStore
+     * 
+     * @return
+     * @throws Exception
+     */
+    boolean deleteAll() throws Exception;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)